NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -181,7 +184,10 @@ class NeuroID private constructor(
         }
     }
 
-    fun setUserID(userId: String): Boolean {
+    fun setUserID(userId: String, checkForEmpty: Boolean = true): Boolean {
```

Comment:
What's the purpose of checkForEmpty?

Reply:
I think he is using it because in the startSession he doesn't care if its empty but normally we do and will reject if they send and empty string. 

However @kishisaka-nid - I think we should rethink this because we don't want to expose a public API that the user could send in a true and an empty string. We only want a setUserID that takes in a string and that's it. Happy to jump on a zoom and brainstorm if you'd like

Reply:
setUserID() is used both externally by apps and internally by startSession(). startSession() will set a random id if the userid is empty, setUserID will fail and do nothing if id is empty/invalid. the checkForEmpty flag is needed to distinguish between the two cases.

Reply:
yeah good point. ill make it separate.

Reply:
Yea that was my assumption, I just saw it was on public api which was why I was asking. Could keep state in memory if they passed in a user Id, which is how i implemented in iOS https://github.com/Neuro-ID/neuroid-ios-sdk/blob/main/NeuroID/NeuroIDClass/Extensions/NIDSession.swift#L128